sklearn/cluster/k_means_.py

```diff
@@ -1415,3 +1416,26 @@ def predict(self, X):
 
         X = self._check_test_data(X)
         return self._labels_inertia_minibatch(X)[0]
+   
+    def _reset_estimator(self):
+        """Forget any previously fit model"""
+        # NOTE: This is faster compared to hasattr, then delattr
```

Comment:
ok :) Do I also revert back to the more readable(?) form... as in https://github.com/ragv/scikit-learn/commit/926f59634a23c27e29fa88e5352ca4057ded68ba ?